pkgs/development/libraries/science/math/suitesparse/default.nix

```
@@ -35,35 +37,69 @@ stdenv.mkDerivation rec {
     mpfr
   ] ++ lib.optional enableCuda cudatoolkit;
 
+  propagatedBuildInputs = [
+    gmp
+    mpfr
+  ];
+
   preConfigure = ''
     # Mongoose and GraphBLAS are packaged separately
     sed -i "Makefile" -e '/GraphBLAS\|Mongoose/d'
   '';
 
+  FC = "${gfortran}/bin/gfortran";
+
+  dontUseCmakeConfigure = true;
+
+  cmakeFlags = [
+    "-DBLA_VENDOR=Generic"
+    # Definitions from CMake's setup hook
+    # Unfortunately, it is not possible to extract the additional cmakeFlags generated there
+    "-DCMAKE_INSTALL_BINDIR=${placeholder "out"}/bin"
+    "-DCMAKE_INSTALL_INCLUDEDIR=${placeholder "dev"}/include"
+    "-DCMAKE_INSTALL_LIBDIR=${placeholder "out"}/lib"
+  ] ++ lib.optionals blas.isILP64 [
+    "-DALLOW_64BIT_BLAS=ON"
+  ];
+
+  # CMAKE_OPTIONS will be picked up via the Makefile
+  CMAKE_OPTIONS = lib.concatStringsSep " "cmakeFlags;
+
   makeFlags = [
-    "INSTALL=${placeholder "out"}"
-    "INSTALL_INCLUDE=${placeholder "dev"}/include"
     "JOBS=$(NIX_BUILD_CORES)"
-    "MY_METIS_LIB=-lmetis"
-  ] ++ lib.optionals blas.isILP64 [
-    "CFLAGS=-DBLAS64"
   ] ++ lib.optionals enableCuda [
     "CUDA_PATH=${cudatoolkit}"
     "CUDART_LIB=${cudatoolkit.lib}/lib/libcudart.so"
     "CUBLAS_LIB=${cudatoolkit}/lib/libcublas.so"
   ] ++ lib.optionals stdenv.isDarwin [
     # Unless these are set, the build will attempt to use `Accelerate` on darwin, see:
     # https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/v5.13.0/SuiteSparse_config/SuiteSparse_config.mk#L368
-    "BLAS=-lblas"
-    "LAPACK=-llapack"
-  ]
-  ;
+    #"BLAS=-lblas"
+    #"LAPACK=-llapack"
+  ];
 
   buildFlags = [
     # Build individual shared libraries, not demos
     "library"
+    "docs"
   ];
 
+  postInstall = ''
+    mkdir -p $doc
+    for docdir in */Doc; do
+      local name="$(dirname "$docdir")"
+      local dest_dir="$doc/share/doc/$name"
+
+      if [[ $name = Mongoose || $name = GraphBLAS ]]; then
+        # Mongoose and GraphBLAS are packaged separately
+        continue
+      fi
```

```suggestion
      if [[ $name = GraphBLAS ]]; then
        # GraphBLAS are packaged separately
        continue
      fi
```

We have mongoose at `2.0.4`, while suitesparse has `3.0.4`. I guess we shouldn't package mongoose separately anymore.

---

>We have mongoose at 2.0.4

that is the latest available version https://github.com/ScottKolo/Mongoose/releases

>while suitesparse has 3.0.4

therefore that seem strange. do they use a fork or something?

---

> therefore that seem strange. do they use a fork or something?

I think the standalone repo is actually a fork [for publication on ACM TOMS](https://github.com/ScottKolo/Mongoose/releases/tag/v2.0.4).

---

The maintainer clarified:

>The version of Mongoose inside the SuiteSparse repo is the primary version now. I've already taken over its maintenance.


https://github.com/DrTimothyAldenDavis/SuiteSparse/issues/25#issuecomment-1491888063